packages/flutter/lib/src/material/paginated_data_table.dart

```diff
@@ -456,7 +477,7 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
       Text(
         localizations.pageRowsInfoTitle(
           _firstRowIndex + 1,
-          _firstRowIndex + widget.rowsPerPage,
+          (_firstRowIndex + widget.rowsPerPage) > _rowCount ? _rowCount : (_firstRowIndex + widget.rowsPerPage),
```

Comment:
Can you split this out into a separate pull request? Trying to fix multiple things with this widget is great! However, it would be a shame if one part of this PR was reverted because the other part created an unforeseen issue. :)

Reply:
Sure I will create a new PR for the new parameters, I didn't thought about this. Thanks

Reply:
Could this be a min function?

```suggestion
          math.min(_firstRowIndex + widget.rowsPerPage, _rowCount),
```